libbpf-tools/klockstat.bpf.c

```diff
@@ -147,6 +148,10 @@ static void account(struct lockholder_info *li)
 {
 	struct lock_stat *ls;
 	u64 delta;
+	u32 key = li->stack_id;
+
+	if (per_thread)
+		key = li->task_id;
```

Comment:
there's a slight risk of running out of space in the stat_map if you run klockstat too long, since there's no reaping of old tasks.  (and you don't want to, since you want the info from whoever was running but exited).  but not worth worrying about.

also, if the user is running with -R (reset after printing), it *will* reap.  so this would only be an issue if there were max_entries processes per output interval.  (which can happen).

Reply:
Right, maybe it's worth add a counter how many it missed due to the lack of storage.